### ISSUE TITLE
🚀 Feature:  Add Semaphore for Messaging API Provider

### ISSUE THREAD

**Issue by moshOntong-IT** [appwrite/appwrite#7714]
### 🔖 Feature description

[Semaphore](https://semaphore.co/) is a SMS API based in PHILIPPINES, it sends bulk SMS or OTP to all Philippines mobile networks such as Globe, Smart, Sun at the cheapest price.

### 🎤 Pitch

In my case, as a startup business, this sms provider is the cheapest pricing offer based on all I have been research.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by eldadfux** [comment]
Thank you for making this suggestion. We're always looking to expand our integrations and add more providers to our different products.

If anyone else would like to see this issue being prioritized please vote on it with a 👍.

If you'd like to pick up this issue, adding new adapters to Appwrite is a great way to start contributing code to the project. For messaging, you can follow our [contribution guide](https://github.com/utopia-php/messaging/blob/main/CONTRIBUTING.md) and this tutorial for adding [a new messaging provider](https://github.com/utopia-php/messaging/blob/main/docs/add-new-adapter.md). 

Before starting to work on this issue please use the comments and asked to be assigned for it to avoid conflicts.

**Comment by 7777chaitanya** [comment]
I would like to take up this issue🙂. Could you assign this to me, please?

**Comment by moshOntong-IT** [comment]
> I would like to take up this issue if @moshOntong-IT doesn't🙂. Could you assign this to me, please?
> 
> 
> 
> Also, the branch naming convention in utopia-php/messaging requires us to add _issue id_. Should I create a duplicate of this issue in utopia-php/messaging as well?

Thanks for taking this feature request 🫶

**Comment by stnguyen90** [comment]
@7777chaitanya, thanks for your interest! 🙏 I’ve assigned the issue to you. Please keep us updated on your progress. If there isn’t any activity after a week, we may unassign you so that someone else can pick up the issue.

> Should I create a duplicate of this issue in utopia-php/messaging as well?

You can use this issue ID.

**Comment by 7777chaitanya** [comment]
Thanks for assigning this to me dear @stnguyen90 🙏🏻. I will keep you posted on my progress. I will make sure to use this issue id.

**Comment by 7777chaitanya** [comment]
Dear Steven, I wanted to provide you with an update regarding the ongoing issue. Unfortunately, I've encountered a roadblock in the resolution process. In an effort to seek assistance and insights, I have reached out to the community. You can find the details in this [support post](https://discord.com/channels/564160730845151244/1216373878499049512). Thank you!

**Comment by sehgalaayu** [comment]
I’d like to work on this issue and implement the Semaphore provider integration following the current messaging provider architecture. Please treat this as claimed by me unless there is active parallel work.
